nexus_capabilities/src/capabilities/transport_amr.hpp

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright (C) 2024 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef NEXUS_WORKCELL_ORCHESTRATOR__TRANSPORT_AMR_HPP
+#define NEXUS_WORKCELL_ORCHESTRATOR__TRANSPORT_AMR_HPP
+
+#include <nexus_capabilities/context_manager.hpp>
+#include <nexus_common/sync_service_client.hpp>
+
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_lifecycle/lifecycle_node.hpp>
+
+#include <rmf_dispenser_msgs/msg/dispenser_request.hpp>
+#include <rmf_dispenser_msgs/msg/dispenser_result.hpp>
+#include <rmf_task_msgs/msg/api_request.hpp>
+#include <rmf_task_msgs/msg/api_response.hpp>
+
+#include <behaviortree_cpp_v3/action_node.h>
+
+#include <deque>
+
+namespace nexus::capabilities {
+
+struct AmrDestination {
```

Comment:
Make `std::variant<rmf_dispenser_msgs::DispenserRequest, rmf_ingestor_msgs::IngestorRequest>` and rename to `Destination`?

Then we have separate capabilities to construct `DispenserRequest` and `IngestorRequest` from string and return a vector needed by the DispatchRmfRequest skill.